src/NUnitFramework/tests/Internal/Filters/FullNameFilterTests.cs

```diff
@@ -61,13 +62,51 @@ public void PassTest()
             Assert.False(_filter.Pass(_anotherFixture));
         }
 
+        [Test]
         public void ExplicitMatchTest()
         {
             Assert.That(_filter.IsExplicitMatch(_topLevelSuite));
             Assert.That(_filter.IsExplicitMatch(_dummyFixture));
-            Assert.False(_filter.IsExplicitMatch(_dummyFixture.Tests[0]));
+            if (((FullNameFilter)_filter).IsRegex)
+                Assert.That(_filter.IsExplicitMatch(_dummyFixture.Tests[0]));
+            else
+                Assert.False(_filter.IsExplicitMatch(_dummyFixture.Tests[0]));
 
             Assert.False(_filter.IsExplicitMatch(_anotherFixture));
         }
+
+        [Test]
+        public void UTF8CharacterMatchFromXML()
+        {
+            TestFilter filter = TestFilter.FromXml("<filter><test>NUnit.Framework.Internal.Filters.FullNameFilterTests.UTF8CharacterMatchFromXMLTestMethod(&quot;ABC\u0001&quot;)</test></filter>");
+            var test = TestBuilder.MakeTestFromMethod(GetType(), nameof(UTF8CharacterMatchFromXMLTestMethod));
+
+            Assert.IsTrue(test.HasChildren);
+            Assert.IsTrue(test.Tests[0].Arguments[0] is string);
+            Assert.IsTrue((string)test.Tests[0].Arguments[0] == "ABC\x01");
+            Assert.IsTrue(filter.Match(test.Tests[0]));
+        }
+
+        [Test]
+        public void ArgumentsMatch()
+        {
+            TestFilter filter = TestFilter.FromXml("<filter><test>NUnit.Framework.Internal.Filters.FullNameFilterTests.ArgumentsMatchTestMethod(1)</test></filter>");
+            var test = TestBuilder.MakeTestFromMethod(GetType(), nameof(ArgumentsMatchTestMethod));
+
+            Assert.IsTrue(test.HasChildren);
+            Assert.IsTrue(test.Tests[0].Arguments[0] is int);
+            Assert.IsTrue((int)test.Tests[0].Arguments[0] == 1);
+            Assert.IsTrue(filter.Match(test.Tests[0]));
+        }
+
+        [TestCase("ABC\x01")]
+        public void UTF8CharacterMatchFromXMLTestMethod(string s)
```

Comment:
All .NET strings are UTF16 and it doesn't make sense to talk about UTF8 characters within them. `"\x01"` is a control character.